sunpy/net/_attrs.py

```diff
@@ -35,26 +35,25 @@ class Time(Range):
         functionality.
 
     """
+    type_name = "time"
 
     def __init__(self, start, end=None, near=None):
         if end is None and not isinstance(start, TimeRange):
             raise ValueError("Specify start and end or start has to be a TimeRange")
         if isinstance(start, TimeRange):
-            self.start = start.start
-            self.end = start.end
+            self.start, self.end = start.start, start.end
         else:
-            self.start = parse_time(start)
-            self.end = parse_time(end)
+            self.start, self.end = parse_time(start), parse_time(end)
 
         if self.start > self.end:
             raise ValueError("End time must be after start time.")
-        self.near = None if near is None else parse_time(near)
+        self.near = parse_time(near) if near is not None else None
 
         super().__init__(self.start, self.end)
 
     def __hash__(self):
-        if not (isinstance(self.start, collections.abc.Hashable) and
-                isinstance(self.end, collections.abc.Hashable)):
+        if not isinstance(self.start, collections.abc.Hashable) or \
```

Comment:
I am not a fan of line break characters. 
Also what the reason to change this from "and" to an "or"?

Reply:
In my opinion, I prefer seen boolean statements simplified via De Morgan laws instead of layers of not and logic conjunctions; I find them more readable

Reply:
If not split the statement will cause the linter to complain, How can I overcome the problem?

Reply:
What are the De Morgan laws?
You can ignore the linter.

Reply:
[Here](https://nerderati.com/2018/06/27/leverage-demorgans-laws-to-rewrite-boolean-expressions/) is a nice practical article and if you are interested in the theoretical aspect of it [this](https://en.wikipedia.org/wiki/De_Morgan's_laws) is the Wikipedia page